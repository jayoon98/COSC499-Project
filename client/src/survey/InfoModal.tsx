import React from 'react';
import { View, StyleSheet, Text, Modal } from 'react-native';
import { ScrollView } from 'react-native-gesture-handler';
import { Button } from '../common/Core';

export type InfoModalProps = {
  // Long description for a question
  description: string;
  visible?: boolean;
  onClose: () => void;
};

export function InfoModal({ description, visible, onClose }: InfoModalProps) {
  return (
    <Modal
      animationType="slide"
      transparent={true}
      visible={visible}
      onDismiss={onClose}
    >
      <View style={styles.container}>
        <View style={styles.modalView}>
          <ScrollView>
            <Text style={{ fontSize: 20 }}>{description}</Text>
          </ScrollView>
          <Button onPress={onClose} style={{ width: '100%', marginTop: 20 }}>
            <Text>close</Text>
          </Button>
        </View>
      </View>
    </Modal>
  );
}

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    // 25% is the size of the header (doubled to use as margin)
    // +5% for exta margin.
    marginTop: '55%',
    justifyContent: 'center',
  },
  modalView: {
    display: 'flex',
    justifyContent: 'space-between',
    margin: 20,
    backgroundColor: 'white',
    borderRadius: 20,
    padding: 22,
    minHeight: 180,
    maxHeight: 500,
    alignItems: 'center',
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
});
