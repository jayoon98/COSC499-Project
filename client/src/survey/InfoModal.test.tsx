import React from 'react';
import { View } from 'react-native';
import renderer from 'react-test-renderer';
import { InfoModal } from './InfoModal';

jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));

it('completed surveys renders correctly', () => {
  const tree = renderer
    .create(
      <View>
        <InfoModal description={'bla bla'} onClose={() => null} />
      </View>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
