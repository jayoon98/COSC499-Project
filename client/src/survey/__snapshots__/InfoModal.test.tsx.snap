// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`completed surveys renders correctly 1`] = `
<View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onDismiss={[Function]}
    transparent={true}
    visible={true}
  >
    <View
      style={
        Object {
          "display": "flex",
          "justifyContent": "center",
          "marginTop": "55%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 20,
            "display": "flex",
            "elevation": 5,
            "justifyContent": "space-between",
            "margin": 20,
            "maxHeight": 500,
            "minHeight": 180,
            "padding": 22,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
          }
        }
      >
        <RCTScrollView
          collapsable={false}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
        >
          <View>
            <Text
              style={
                Object {
                  "fontSize": 20,
                }
              }
            >
              bla bla
            </Text>
          </View>
        </RCTScrollView>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginTop": 20,
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#e2e4ef",
                "borderRadius": 4,
                "borderWidth": 1,
                "display": "flex",
                "justifyContent": "center",
                "padding": 12,
              }
            }
          >
            <Text>
              close
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
